lib/oauth2/request.ex

```diff
@@ -33,16 +34,20 @@ defmodule OAuth2.Request do
       """)
     end
 
-    case :hackney.request(method, url, headers, body, req_opts) do
-      {:ok, ref} when is_reference(ref) ->
-        {:ok, ref}
-
-      {:ok, status, headers, ref} when is_reference(ref) ->
-        process_body(client, status, headers, ref)
-
-      {:ok, status, headers, body} when is_binary(body) ->
+    case Tesla.request(http_client(),
+           method: method,
+           url: url,
+           query: params,
+           headers: headers,
+           body: body,
+           opts: [adapter: req_opts]
+         ) do
+      {:ok, %{status: status, headers: headers, body: body}} when is_binary(body) ->
         process_body(client, status, headers, body)
 
+      {:ok, %{body: ref}} when is_reference(ref) ->
+        {:ok, ref}
```

Comment:
Do you mind expanding on this? It does not match the previous behavior

Reply:
This is because Tesla is doing most of the heavy lifting of references in hackney but if you specify that you really do want the reference like the tests do we return the same things as would be returned while we were only using hackney.

Reply:
https://github.com/ueberauth/oauth2/pull/170/files#diff-afe0a3dcb5431812fea907ffba772b359b0c365bf7f70bacb10082286e72498bR201